tirocks/src/db/imp.rs

```diff
@@ -0,0 +1,726 @@
+    ) -> Self {
+        Self {
+            ptr,
+            _env: env,
+            comparator,
+            handles,
+            is_titan,
+        }
+    }
+
+    fn clear_handles(&mut self) -> Result<()> {
+        for mut h in self.handles.drain(..) {
+            unsafe {
+                h.maybe_drop(self.ptr)?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Close the DB by releasing resources, closing files etc. This will not fsync the WAL files.
+    /// If syncing is required, the caller must first call SyncWAL(), or Write() using an empty
+    /// write batch with WriteOptions.sync=true.
+    ///
+    /// Unlike C++ API, if the close is aborted due to unreleased snapshot, the DB will be dropped
+    /// any way as part of its resource has been freed already.
+    pub fn close(mut self) -> Result<()> {
+        unsafe {
+            self.clear_handles()?;
+            ffi_call!(crocksdb_close(self.ptr))
+        }
+    }
+
+    /// ListColumnFamilies will open the DB specified by argument name and return the list of all
+    /// column families in that DB. The ordering of returned column families is unspecified.
+    pub fn list_cfs(db: DbOptions, path: impl AsRef<Path>) -> Result<Vec<String>> {
+        let mut convert_s = Status::default();
+        let mut names = Vec::new();
+        unsafe {
+            let mut handle = |name: &[u8]| match str::from_utf8(name) {
+                Ok(n) => names.push(n.to_string()),
+                Err(e) => {
+                    convert_s = Status::with_error(Code::kCorruption, format!("{}", e));
+                }
+            };
+            let (ctx, fp) = util::wrap_string_receiver(&mut handle);
+            ffi_call!(crocksdb_list_column_families(
+                db.get_ptr(),
+                path.as_ref().path_to_slice(),
+                ctx,
+                Some(fp),
+            ))?;
+        }
+        if !convert_s.ok() {
+            return Err(convert_s);
+        }
+        Ok(names)
+    }
+
+    /// Create a column_family.
+    ///
+    /// If it's a titan db, it's the same as calling `create_cf_titan` with the titan specific
+    /// options set to default.
+    pub fn create_cf(&mut self, name: impl AsRef<str>, opt: CfOptions) -> Result<()> {
+        if self.is_titan {
+            return self.create_cf_titan(name, opt.into());
+        }
+        let ptr = unsafe {
+            ffi_call!(crocksdb_create_column_family(
+                self.ptr,
+                opt.as_ptr(),
+                r(name.as_ref().as_bytes()),
+            ))
+        }?;
+        if let Some(c) = opt.comparator() {
+            self.comparator.push(c.clone());
+        }
+        self.handles
+            .push(unsafe { RefCountedCfHandle::from_ptr(ptr, true) });
+        Ok(())
+    }
+
+    /// Create a column_family.
+    ///
+    /// If it's not titan, it's the same as calling `create_cf`. Titan specific options will be
+    /// ignored.
+    pub fn create_cf_titan(
```

Comment:
how about only publicizing `create_cf`

Reply:
Then there is no way to customize the cf options for titan